### ISSUE TITLE
"Add type annotation" is missing "dynamic" resulting in invalid code

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#33609]
I've seen this happen a couple of times in the Flutter project. Given code like this:

```dart
Future<Map<String, dynamic>> test() async {
  return null;
}

Future<Map<String, dynamic>> test2() async {
  final a = await test();
  return a;
}
```

If you invoke `Add type annotation` on the `final a = ...` line, you end up with code like this:

```dart

Future<Map<String, dynamic>> test() async {
  return null;
}

Future<Map<String, dynamic>> test2() async {
  final Map<String, > a = await test();
  return a;
}
```

For some reason the word `dynamic` has been omitted so you get an error and have to add it in manually. I can't see any obvious reason why this wouldn't work.

**Comment by bwilkerson** [comment]
That's a bug. There's code to prevent adding `dynamic` at the top level, but it appears that we're not distinguishing between top-level and nested uses.

**Comment by DanTup** [comment]
This came up in Gitter today, I think it's the same cause. Given this code:

```dart
class ClassA<T> {}

class ClassB<A, B extends ClassA<A>> {}

abstract class ClassC {
  List<ClassB> get list;
}

class ClassD extends ClassC{
}
```

If you invoke "Create 1 missing override" on `ClassD`, you get this:

```dart
List<ClassB<, ClassA>> get list => null;
```

Since no type was given for `A` in the definition of ClassC, it's presumably inferred as dynamic, but then missing from the generated code.

**Comment by DanTup** [comment]
This appears to have been fixed (probably by fdf4a1cc478bb4f3c7820b7b7ea4fa96ed23a585, which has a test similar to the code above and adds `required: true` when writing type args).
